test/integration_test/snapshot_restore_test.go

```diff
@@ -0,0 +1,91 @@
+// +build integrationtest
+
+package integrationtest
+
+import (
+	"testing"
+	"time"
+
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/stretchr/testify/require"
+)
+
+func testSnapshotRestore(t *testing.T) {
+	t.Run("simpleSnapshotRestoreTest", simpleSnapshotRestoreTest)
+	t.Run("groupSnapshotRestoreTest", groupSnapshotRestoreTest)
+	t.Run("cloudSnapshotRestoreTest", cloudSnapshotRestoreTest)
+	t.Run("groupCloudSnapshotRestoreTest", groupCloudSnapshotRestoreTest)
+}
+
+func createInPlaceRestore(t *testing.T, namespace string, appKeys []string) []*scheduler.Context {
+	ctxs, err := schedulerDriver.Schedule(namespace,
+		scheduler.ScheduleOptions{AppKeys: appKeys})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+	return ctxs
+}
+
+func verifyInPlaceSnapshotRestore(t *testing.T, ctx []*scheduler.Context, startTime time.Time, timeout time.Duration) {
+	err := schedulerDriver.WaitForRunning(ctx[0], timeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for task")
+	err = schedulerDriver.ValidateVolumeSnapshotRestore(ctx[0], startTime)
+	require.NoError(t, err, "Error validating volumesnapshotRestore")
+}
+
+func simpleSnapshotRestoreTest(t *testing.T) {
+	// foldername + name as namespaces
+	snapCtx, err := schedulerDriver.Schedule("splocal",
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-snap-restore"}})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(snapCtx), "Only one task should have started")
+	verifySnapshot(t, snapCtx, "mysql-data", defaultWaitTimeout)
+	startTime := time.Now()
+	restoreCtx := createInPlaceRestore(t, "splocal", []string{"mysql-snap-inplace-restore"})
+	verifyInPlaceSnapshotRestore(t, restoreCtx, startTime, defaultWaitTimeout)
+
+	// cleanup test
+	destroyAndWait(t, snapCtx)
+	destroyAndWait(t, restoreCtx)
+}
+
+func groupSnapshotRestoreTest(t *testing.T) {
+	snapCtx, err := schedulerDriver.Schedule("spgroup",
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-localsnap-group"}})
+	require.NoError(t, err, "Error scheduling task")
+	//require.Len(t, snapCtx, 1, "Only one task should have started")
+
+	for _, ctx := range snapCtx {
+		verifyGroupSnapshot(t, ctx, defaultWaitTimeout)
```

Comment:
Why are there multiple contexts for this?

Reply:
yes, does not required